SoObjects/Mailer/SOGoMailBaseObject.m

```diff
@@ -206,6 +220,22 @@ - (NGImap4Connection *) imap4Connection
       [imap4 retain];
     }
 
+   // Connection broken, try to reconnect
+   if (![[[imap4 client] isConnected] boolValue])
```

Comment:
Agreed on first change, but I don't understand the second one, wouldn't imap4Connection call raise the exception before reaching your code?